packages/core/src/combinator/variadic.ts

```diff
@@ -1,9 +1,7 @@
 import { Stream } from '@most/types'
 
-// Map arrays to arrays of Streams:
-// Array<A> => Array<Stream<A>>
-// [A, B, C, ...] => [Stream<A>, Stream<B>, Stream<C>, ...]
-// TODO: use readonly any[] once TS 3.4.x has been in the wild for "enough" time
-export type ToStreamsArray<A extends ArrayLike<any>> = {
+type LiteralArray<A extends ArrayLike<any>> = {
   [K in keyof A]: Stream<A[K]>
 }
+
+export type InputStreamArray<Args extends ArrayLike<any>> = Readonly<LiteralArray<Args> | Stream<unknown>[]>
```

Comment:
@nissoh I'm not sure I understand why we need `| Stream<unknown>[]` here.  Can you say a bit more about the use cases it helps?  Thanks!

Reply:
`| Stream<unknown>[]`, serve as compatibility to dynamic arrays. without it, dynamic input of dynamic arrays throws type error

There might be a better and understandable solution with typescript version 4

Reply:
Do we have a test for dynamic arrays?